pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -80,9 +84,9 @@
   private final long _startTimeMillis = System.currentTimeMillis();
 
   private final String _segmentName;
-  private final Schema _schema;
+  private final Schema _originalSchema;
   private final int _capacity;
-  private final SegmentMetadata _segmentMetadata;
+  private final SegmentMetadata _originalSegmentMetadata;
```

Comment:
Same here. Why rename?

Reply:
same above, segmentMetadata is immutable, just to be super clear it does not reflect the newly added columns.

Reply:
Sounds good. I have reverted the rename